db/view/view.cc

```diff
@@ -1842,6 +1849,28 @@ get_view_natural_endpoint(
         process_candidate(base_endpoints, topology, base_endpoint, false);
     }
 
+    // if we're a pending base replica and we're ready for reading then we should generate view updates same as
```

Comment:
@wmitros Please review the PR as well, you should have a better understanding of `get_view_natural_endpoint`